src/suites/cts/validation/createBindGroup.spec.ts

```diff
@@ -0,0 +1,491 @@
+export const description = `
+createBindGroup validation tests.
+`;
+
+import { TestGroup } from '../../../framework/index.js';
+
+import { ValidationTest } from './validation_test.js';
+
+function clone(descriptor: GPUTextureDescriptor): GPUTextureDescriptor {
+  return JSON.parse(JSON.stringify(descriptor));
+}
+
+export class F extends ValidationTest {
+  getStorageBuffer(): GPUBuffer {
+    return this.device.createBuffer({
+      size: 1024,
+      usage: GPUBufferUsage.STORAGE,
+    });
+  }
+
+  getUniformBuffer(): GPUBuffer {
+    return this.device.createBuffer({
+      size: 1024,
+      usage: GPUBufferUsage.UNIFORM,
+    });
+  }
+
+  getSampler(): GPUSampler {
+    return this.device.createSampler();
+  }
+
+  getSampledTexture(): GPUTexture {
+    return this.device.createTexture({
+      size: { width: 16, height: 16, depth: 1 },
+      format: 'rgba8unorm',
+      usage: GPUTextureUsage.SAMPLED,
+    });
+  }
+
+  getStorageTexture(): GPUTexture {
+    return this.device.createTexture({
+      size: { width: 16, height: 16, depth: 1 },
+      format: 'rgba8unorm',
+      usage: GPUTextureUsage.STORAGE,
+    });
+  }
+
+  async getErrorBuffer(): Promise<GPUBuffer> {
```

Comment:
nit: this will be useful for many tests and all error buffers are idempotent so it could go on ValidationTest.